accounts-db/src/tiered_storage/owners.rs

```diff
@@ -40,16 +41,16 @@ pub enum OwnersBlockFormat {
 
 impl OwnersBlockFormat {
     /// Persists the provided owners' addresses into the specified file.
-    pub fn write_owners_block(
+    pub(crate) fn write_owners_block<'a>(
         &self,
         file: &TieredStorageFile,
-        addresses: &[Pubkey],
+        owners: impl IntoIterator<Item = &'a &'a Pubkey>,
```

Comment:
It seems that as I use `&Pubkey` as key to avoid copying, it also results in an odd Iterator type 😂

Reply:
Directly passing OwnersTable instead.